stacks-common/build.rs

```diff
@@ -1,8 +1,64 @@
 use std::path::Path;
+use std::process::Command;
 use std::{env, fs};
 
 use toml::Value;
 
+fn current_git_hash() -> Option<String> {
+    if option_env!("GIT_COMMIT") == None {
+        let commit = Command::new("git")
+            .arg("log")
+            .arg("-1")
+            .arg("--pretty=format:%h") // Abbreviated commit hash
+            .current_dir(env!("CARGO_MANIFEST_DIR"))
+            .output();
+
+        if let Ok(commit) = commit {
+            if let Ok(commit) = String::from_utf8(commit.stdout) {
+                return Some(commit.trim().to_string());
+            }
+        }
+    } else {
+        return option_env!("GIT_COMMIT").map(String::from);
+    }
+
+    None
+}
+
+fn current_git_branch() -> Option<String> {
+    if option_env!("GIT_BRANCH") == None {
+        let commit = Command::new("git")
+            .arg("rev-parse")
+            .arg("--abbrev-ref")
+            .arg("HEAD")
+            .output();
+        if let Ok(commit) = commit {
+            if let Ok(commit) = String::from_utf8(commit.stdout) {
+                return Some(commit.trim().to_string());
+            }
+        }
+    } else {
+        return option_env!("GIT_BRANCH").map(String::from);
+    }
+
+    None
+}
```

Comment:
How about this instead?

```suggestion
fn current_git_branch() -> Option<String> {
    option_env!("GIT_BRANCH")
        .map(String::from)
        .or_else(|| {
            Command::new("git")
                .arg("rev-parse")
                .arg("--abbrev-ref")
                .arg("HEAD")
                .output()
                .map(|output| String::from_utf8(output.stdout))
                .ok()
        })
}
```